### PR TITLE
fix: Language specific fields not suffixed by a language should be in the language of the interface

### PR DESCRIPTION
**Pull request by stephanegigandet**
When an app sends ingredients_text, it should be assumed that it is in the language of the interface (e.g. ingredients_text_en) and should not be kept in ingredients_text if the language of the product is different from the language of the interface.

Should fix some test differences in https://github.com/openfoodfacts/openfoodfacts-server/pull/8855

### CODE REVIEW

File: tests/integration/api_v2_product_write.t [hunk 1]
```diff
@@ -103,6 +103,29 @@ my $tests_ref = [
...
 	},
+	# Test that we use the language of the interface (lc) for language fields without a language suffix
+	{
+		test_case => 'post-product-ingredients-text-without-language',
+		method => 'POST',
+		path => '/cgi/product_jqm_multilingual.pl',
+		form => {
+			cc => "uk",
+			lc => "en",
+			lang => "fr",
```
Comment (alexgarel) [inline tests/integration/api_v2_product_write.t:114]: A comment here to explain this is welcome: what is the difference between lc and cc (I'm always lost)

File: tests/integration/expected_test_results/search_v1/search-tags-categories-without-ingredients-from-palm-oil.json [hunk 2]
```diff
@@ -1,740 +1,8 @@
 {
-   "count" : 1,
+   "count" : 0,
```
Comment (alexgarel) [inline tests/integration/expected_test_results/search_v1/search-tags-categories-without-ingredients-from-palm-oil.json:2]: Why this change ?
Is the test still valid so ?

File: tests/integration/api_v2_product_write.t [hunk 3]
```diff
@@ -103,6 +103,29 @@ my $tests_ref = [
...
+			quantity => "250 g",
+			serving_size => '20 g',
+			ingredients_text => "Pork meat, salt",
+		}
+	},
+	{
+		test_case => 'get-product-ingredients-text-without-language',
+		method => 'GET',
+		path => '/api/v2/product/1234567890004',
+		expected_status_code => 404,
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply tests/integration/api_v2_product_write.t:127]: the status code should be 200 in fact, wrong copy/paste

I will add some comments for cc / lc / lang

[the remark this answers is not in the document]
Comment (stephanegigandet) [reply tests/integration/api_v2_product_write.t:127]: no parameter is mandatory: cc (country of the interface) and lc (language of the interface) will have default values if not provided ("world" and "en"), and lang is the main language of the product (for a new product, if not provided, we will fallback to the value of lc)

File: tests/integration/expected_test_results/search_v1/search-no-filter.json [hunk 4]
```diff
@@ -1832,23 +1832,14 @@
       {
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply tests/integration/expected_test_results/search_v1/search-no-filter.json:1832]: We have those differences because of the bug: we were setting "ingredients_text" with English ingredients while the main language of the product (lang field) was Spanish, so we analyzed (wrongly) ingredients, and set the ingredients analysis fields.

With this correction, we instead set ingredients_text_en, (and not ingredients_text), and we don't analyze ingredients because we don't have them in the main language of the product.

@raphael0202 's PR https://github.com/openfoodfacts/openfoodfacts-server/pull/8855 will then change those tests again: we will detect we have ingredients in English, and analyze them.

**Review by raphael0202 — APPROVED** [review]
